dhis-2/dhis-services/dhis-service-administration/src/main/java/org/hisp/dhis/resourcetable/DefaultResourceTableService.java

```diff
@@ -255,4 +258,29 @@ public void dropAllSqlViews( JobProgress progress )
         progress.startingStage( "Drop SQL views", nonQueryViews.size(), SKIP_ITEM );
         progress.runStage( nonQueryViews, SqlView::getViewName, sqlViewService::dropViewTable );
     }
+
+    @Override
+    public List<Integer> generateDataYears()
+    {
+        List<Integer> availableDataYears = resourceTableStore.getAvailableDataYears();
+
+        if ( availableDataYears.isEmpty() )
+        {
+            availableDataYears = new ArrayList<>();
+            availableDataYears.add( now().getYear() );
+        }
+
+        int firstYear = availableDataYears.get( 0 );
+        int lastYear = availableDataYears.get( availableDataYears.size() - 1 );
+
+        for ( int i = 0; i < 5; i++ )
```

Comment:
I believe would be nice to extract the number `5` into a constant to give it a better context.

Reply:
this is a good point, done.